### ISSUE TITLE
Upload wheels for all commits to pip.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1561]
We currently upload wheels for all commits to S3, but it would be even better to put them in pip so that people can do `pip install ray-latest` or something like that. See for example https://pypi.python.org/pypi/tf-nightly. It would also be nice to be able to do something like

```
pip install ray-latest==ab37d0cd1964581e55122ac13bbdf950fad77fce
```

but that might be a little tricker.

**Comment by richardliaw** [comment]
#1606 is a nice workaround for this. One followup would be to expose these wheels in the docs + readme for public consumption.

**Comment by robertnishihara** [comment]
Agreed, it always takes me forever to find the right command.

```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.3.1-cp36-cp36m-manylinux1_x86_64.whl
```

we can provide the analogous command for each wheel, but whenever we bump the version number we'll need to update the wheel names.

**Comment by richardliaw** [comment]
Should be addressed in #1606 and #1672
